app/helpers/embed_player_helper.rb

```diff
@@ -111,19 +92,83 @@ def embed_player_category_options(podcast, selected)
     options_for_select(opts, selected)
   end
 
-  def embed_player_all_episodes_options(selected)
-    opts = %w[all number].map { |v| [t("helpers.label.podcast_player.episodes_options.#{v}"), v] }
-    options_for_select(opts, selected)
-  end
-
   private
 
-  def embed_params(params, preview = false)
+  def embed_params(options_with_defaults = {})
+    opts = options_with_defaults.reject do |key, val|
+      DEFAULT_OPTIONS[key.to_sym] == val
+    end
+
+    # shared params
+    params = {}
+    params[EMBED_PLAYER_CARD] = "1" if opts[:embed_player_type] == "card"
+    params[EMBED_PLAYER_THEME] = opts[:embed_player_theme] if opts[:embed_player_theme].present?
+    params[EMBED_PLAYER_ACCENT_COLOR] = opts[:accent_color].sub("#", "") if opts[:accent_color].present?
+
+    # playlist params
+    if opts[:playlist]
+      params[EMBED_PLAYER_PLAYLIST] = (opts[:episode_number].to_i > 1) ? opts[:episode_number] : "all"
+      params[EMBED_PLAYER_SEASON] = opts[:season] if opts[:season].to_i > 0
+      params[EMBED_PLAYER_CATEGORY] = opts[:category] if opts[:category].to_s.strip.present?
+    end
+
+    params
+  end
+
+  def embed_url(params, preview = false)
     "https://#{ENV["PLAY_HOST"]}#{preview ? EMBED_PLAYER_PREVIEW_PATH : EMBED_PLAYER_PATH}?#{params.to_query}"
   end
 
   def enclosure_with_token(ep)
     sep = ep.enclosure_url.include?("?") ? "&" : "?"
     ep.enclosure_url + sep + {DOVETAIL_TOKEN => prx_jwt}.to_query
   end
+
+  def embed_player_iframe(options, src = "")
+    is_card = options[:embed_player_type] == "card"
+    fixed_width = options[:max_width].to_i if options[:max_width].to_i >= 300
+
+    # calculate height for playlists
+    height =
+      if options[:episode_number].to_i.between?(1, 5)
+        HEIGHT_BASE + HEIGHT_PLAYLIST_HEADER + HEIGHT_PLAYLIST_ROW * options[:episode_number].to_i
+      elsif options[:playlist]
+        (HEIGHT_BASE + HEIGHT_PLAYLIST_HEADER + HEIGHT_PLAYLIST_ROW * 5.5).round
+      else
+        HEIGHT_BASE
+      end
```

Comment:
This method gets in the weeds a bit.  But it's ported from the embed player repo.  And there are advantages to being able to generate these completely in ruby (return API results, programmatically generate for a bunch of episodes, etc).

Reply:
That's not a bad idea.  That would get us away from having users copy/paste iframes with hardcoded styles right?  So we could say, change the heights of things in the play repo.